### PR TITLE
Add Gauges APIs examples

### PR DESCRIPTION
**Pull request by mayurkale22**

### CODE REVIEW

**Comment by songy23** [comment]
Please fix the build.

File: CHANGELOG.md [hunk 1]
```diff
@@ -7,6 +7,7 @@
   if you are using `gke_container` resources. For more info,
   https://cloud.google.com/monitoring/kubernetes-engine/migration#incompatible
 - Add OpenCensus Java OC-Agent Trace Exporter.
+- Add Gauges (`DoubleGauge`, `LongGauge`, `DerivedDoubleGauge`, `DerivedLongGauge`) APIs.
```
Comment (bogdandrutu) [inline CHANGELOG.md]: This should get into a different PR correct?

**Comment by bogdandrutu** [comment]
Can you please update this PR and make it ready for review?

**Comment by mayurkale22** [comment]
Dependency with https://github.com/census-instrumentation/opencensus-java/pull/1557

**Comment by mayurkale22** [comment]
@bogdandrutu PR is ready for review. PTAL

File: examples/src/main/java/io/opencensus/examples/gauges/DerivedLongGaugeQuickstart.java [hunk 2]
```diff
@@ -0,0 +1,65 @@
...
+      Collections.singletonList(LabelValue.create("Inbound"));
+
+  private static final DerivedLongGauge derivedLongGauge =
+      metricRegistry.addDerivedLongGauge("queue_size", "Pending jobs", "1", labelKeys);
+
+  // To instrument a queue's depth.
+  private static void doWork() {
+    derivedLongGauge.createTimeSeries(
+        labelValues,
+        new LinkedBlockingQueue(),
```
Comment (bogdandrutu) [inline examples/src/main/java/io/opencensus/examples/gauges/DerivedLongGaugeQuickstart.java]: Keep an instance of this somewhere else to prove that this should not be the only reference.

**Pull request merged**